### PR TITLE
Explain Statement plan improvement

### LINKED ISSUE #14961: Transparent routing fails when using `EXPLAIN`
Given we have a routed table `routed` that is mapped to `realTable`, a query such as this fail:

```sql
explain select * from routed
```

The reason for this is that Vitess was not changing the table names of the routed table before sending it to MySQL, and `routed` does not exist in MySQL

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description
In order to support `EXPLAIN` on more queries, we need to do a little bit of table name massaging before sending down the `EXPLAIN` query to MySQL. Specifically, we need to handle routed tables correctly, since they don't have the same name in Vitess as they do in the underlying MySQL instance.

## Related Issue(s)
Fixes #14961

## Checklist
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required https://github.com/vitessio/website/pull/1669

### CODE REVIEW

**Comment by harshit-gangal** [comment]
Docs PR: https://github.com/vitessio/website/pull/1669

File: go/vt/vtgate/planbuilder/testdata/dml_cases.json [hunk 1]
```diff
@@ -4410,8 +4410,8 @@
     }
   },
   {
-    "comment": "explain dml without any directive should fail",
-    "query": "explain format=vtexplain delete from user",
+    "comment": "vexplain all dml without any directive should fail",
+    "query": "vexplain all delete from user",
     "plan": "VT09008: vexplain queries/all will actually run queries"
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/testdata/dml_cases.json:4415]: It is not clear to me if `vtexplain all` or just `vtexplain` queries, will actually run queries

**Review by frouioui — APPROVED** [review]
Should we add end-to-end tests for this?

Otherwise, it all looks good to me

File: go/vt/vtgate/planbuilder/testdata/dml_cases.json (hunk 1 above)
Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/testdata/dml_cases.json:4415]: both of them will. only `vexplain plan` does not run it.

**Review by systay — APPROVED** [review]

**Pull request merged**